### PR TITLE
1.4 envoy backports 2019 12 10

### PR DESCRIPTION
**Pull request by jrajahalme**
```release-note
Envoy is updated to release 1.12.2, including important security fixes.
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9742)

### CODE REVIEW

**Comment by jrajahalme** [comment]
never-tell-me-the-odds

**Comment by jrajahalme** [comment]
never-tell-me-the-odds

**Comment by jrajahalme** [comment]
Added missing backport

**Comment by jrajahalme** [comment]
never-tell-me-the-odds

**Comment by jrajahalme** [comment]
test-missed-k8s

File: pkg/envoy/server.go [hunk 1]
```diff
@@ -335,14 +335,14 @@ func (s *XDSServer) AddListener(name string, kind policy.L7ParserType, endpointP
 	listenerConf := proto.Clone(s.listenerProto).(*envoy_api_v2.Listener)
 	if kind == policy.ParserTypeHTTP {
 		listenerConf.FilterChains = append(listenerConf.FilterChains, proto.Clone(s.httpFilterChainProto).(*envoy_api_v2_listener.FilterChain))
-		listenerConf.FilterChains[0].Filters[1].ConfigType.(*envoy_api_v2_listener.Filter_Config).Config.Fields["http_filters"].GetListValue().Values[0].GetStructValue().Fields["config"].GetStructValue().Fields["policy_name"] = &structpb.Value{Kind: &structpb.Value_StringValue{StringValue: endpointPolicyName}}
+		// listenerConf.FilterChains[0].Filters[1].ConfigType.(*envoy_api_v2_listener.Filter_Config).Config.Fields["http_filters"].GetListValue().Values[0].GetStructValue().Fields["config"].GetStructValue().Fields["policy_name"] = &structpb.Value{Kind: &structpb.Value_StringValue{StringValue: endpointPolicyName}}
```
Comment (aanm) [inline pkg/envoy/server.go:338]: I assume these comments are on purpose

**Review by aanm — APPROVED** [review]

**Comment by jrajahalme** [comment]
Deployment of Istio test resources failed due to duplicate port allocation. No visibility to who did the first allocation.

```
The Service "istio-ingressgateway" is invalid: spec.ports[1].nodePort: Invalid value: 31380: provided port is already allocated
```

File: pkg/envoy/server.go (hunk 1 above)
Comment (jrajahalme) [reply pkg/envoy/server.go:338]: They can be removed when I next need to touch this code.

**Comment by jrajahalme** [comment]
test-missed-k8s

**Pull request merged**